### ISSUE TITLE
Database names start with a " " blank

### ISSUE THREAD

**Issue by ghost235** [virtualmin/virtualmin-gpl#460]
|  SYSTEM INFORMATION||
|----------------------|---------------------------|
|  OS type and version |     Debian Linux 9     |
|  Webmin version      |     2.000      |
|  Usermin version     |     1.860     |
|  Virtualmin version  |     7.1-1  Pro       |
|  Cloudmin version    |     9.6  Pro       |
|  Theme version       |     20.00      |
|  Package updates     |      3  package updates are available   |

Hi,

I've the issue that new databases are createt with a leeding blan from the script installers.

![grafik](https://user-images.githubusercontent.com/27903262/193417305-008c8b9e-046e-41fd-84b6-5a3ce6b7e961.png)

If I want to create a DB manual I got.

`Failed to create database : Missing or invalid database name - only letters, numbers and _ are allowed`

The server tamplate looks like this.

![grafik](https://user-images.githubusercontent.com/27903262/193417556-23c89ed4-c8c0-4c40-9caa-44c6738da331.png)

Missed I something? Where can I look/ edit this?

**Comment by iliajie** [comment]
Hello,

Do you expect leading space work in a database name?

**Comment by ghost235** [comment]
Hi, 
it is working for the configs (Wordpress,Nextcloud) with arounding ‘‘.
For mysqldump it won’t work. Also `mysql> use „ database“` give me an error.

**Comment by iliajie** [comment]
> Also `mysql> use „ database“` give me an error.

Try using tildas <code>\`</code> to quote your database name, i.e. <code>\` database\`</code>.

**Comment by ghost235** [comment]
Is might work for this. Maybe also `mysqldump` will work with it.
But assigning to a domain/virtualserver wouldn’t work.

**Comment by iliajie** [comment]
We don't expect it to work.

**Comment by chris001** [comment]
On the server template, do you have a space in front of the "Default database name" or "Prefix for additional databases", like ` ${DOM}` ?

**Comment by ghost235** [comment]
> On the server template, do you have a space in front of the "Default database name" or "Prefix for additional databases", like ` ${DOM}` ?

No.

**Comment by ghost235** [comment]
Can I check the `${DOM}` from console?

**Comment by marclaporte** [comment]
> > On the server template, do you have a space in front of the "Default database name" or "Prefix for additional databases", like ` ${DOM}` ?
> 
> No.

It sure looks like you have a space:
![Screenshot_20230813-164909](https://github.com/virtualmin/virtualmin-gpl/assets/1004261/d2089e42-c15d-4cd5-8554-2955c0221fab)

**Comment by marclaporte** [comment]
Indirectly related: It's not rare to get support requests that originate from an extra space in a copy-paste.

Here are related fixes/discussions to add a warning: 

- https://github.com/cypht-org/cypht/pull/743
- https://gitlab.com/tikiwiki/tiki/-/merge_requests/3491

One of our junior devs is exploring Virtualmin and will attempt to generate a merge request for this. Even if it's not accepted, it's a good way to learn Virtualmin and we'll put it in WikiSuite (which uses Virtualmin)

**Comment by jcameron** [comment]
Yes, please send a PR to precent spaces at the start or end of usernames..

**Comment by iliajie** [comment]
If the data passed through the field doesn't support trailing and/or leading spaces it should be trimmed on the server side upon submit.

So, if we don't really want that trailing space to be there &mdash; it has to be simply truncated upon submit.

It was also a similar issue on Webmin login page, with usernames in the past.

**Comment by jcameron** [comment]
That would also work!

**Comment by marclaporte** [comment]
Ok, our junior dev will explore the various places where this could happen and prepare a proposal.

**Comment by iliaross** [comment]
Reopened, as this was closed by an automated cleanup run.

⚠️ If the issue is resolved, please take a moment to close it to help keep the project organized.
